demisto_sdk/commands/common/hook_validations/base_validator.py

```diff
@@ -26,49 +27,103 @@ class BaseValidator:
 
     def __init__(self, ignored_errors=None, print_as_warnings=False, suppress_print: bool = False,
                  json_file_path: Optional[str] = None):
+        # these are the ignored errors from the .pack-ignore including un-allowed error codes
         self.ignored_errors = ignored_errors if ignored_errors else {}
+        # these are the predefined ignored errors from packs which are partner/community support based.
+        # represented by PRESET_ERROR_TO_IGNORE
+        self.predefined_by_support_ignored_errors = {}  # type: ignore[var-annotated]
+        # these are the predefined ignored errors from packs which are deprecated.
+        # represented by PRESET_ERROR_TO_CHECK
+        self.predefined_deprecated_ignored_errors = {}  # type: ignore[var-annotated]
         self.print_as_warnings = print_as_warnings
         self.checked_files = set()  # type: ignore
         self.suppress_print = suppress_print
         self.json_file_path = json_file_path
 
     @staticmethod
-    def should_ignore_error(error_code, ignored_errors):
-        """Return True is code should be ignored and False otherwise"""
-        if ignored_errors is None:
-            return False
+    def should_ignore_error(
+        error_code,
+        ignored_errors_pack_ignore,
+        predefined_deprecated_ignored_errors,
+        predefined_by_support_ignored_errors
+    ):
+        """
+        Determine if an error should be ignored or not.
+
+        Args:
+            error_code (str): the error code of the validation.
+            ignored_errors_pack_ignore (list[str]): a list of ignored errors which is
+                part of the .pack-ignore that belongs to a specific file.
+            predefined_deprecated_ignored_errors (list[str]): a list of ignored errors which are part of a deprecated
+                content entity.
+            predefined_by_support_ignored_errors (list[str)): a list of ignored errors which are part of the support
+                level of a pack.
+
+        Returns:
+            True if error should be ignored, False if not.
+        """
+        error_type = error_code[:2]
 
-        # check if specific codes are ignored
-        if error_code in ignored_errors:
+        if error_code in predefined_deprecated_ignored_errors or error_type in predefined_deprecated_ignored_errors:
             return True
 
-        # in case a whole section of codes are selected
-        code_type = error_code[:2]
-        if code_type in ignored_errors:
+        if error_code in predefined_by_support_ignored_errors or error_type in predefined_by_support_ignored_errors:
             return True
 
-        return False
+        return (
+            error_code in ignored_errors_pack_ignore or error_type in ignored_errors_pack_ignore
+        ) and (
+            error_code in ALLOWED_IGNORE_ERRORS
+        )
+
+    @staticmethod
+    def should_error_code_not_be_ignored_in_pack_ignore(error_code, ignored_errors_pack_ignore):
```

Comment:
what's the difference between this one and `should_ignore_error`?

Reply:
basically the ```should_error_code_not_be_ignored_in_pack_ignore``` method checks out if an error code that is **in** the .pack-ignore should be ignored or not, while the ```should_ignore_error``` method defines if an error should be ignored in general (including errors from .pack-ignore and errors that can be ignored in parter/community supported/deprecated packs

Reply:
I see, thanks
perhaps a name like `is_error_allowed_in_pack_ignore` (keep in mind to revert the boolean result) will be clearer

Reply:
will be done!